### PR TITLE
Add serverZip property to prepare the server from zip automatically

### PR DESCRIPTION
**Pull request by jmarkos**

### CODE REVIEW

**Comment by rvansa** [comment]
Could you rather do that using `java.util.zip.ZipInputStream` than introducing a random dependency from the internetz?

**Comment by jmarkos** [comment]
Done.

File: plugins/infinispan52/src/main/java/org/radargun/service/InfinispanServerService.java [hunk 1]
```diff
@@ -89,6 +100,42 @@ public void init() {
...
       try {
+         if (serverZip != null) {
+            Path homePath = FileSystems.getDefault().getPath(home);
+            Utils.deleteDirectory(homePath.toFile());
+            Files.createDirectory(homePath);
+
+            ZipInputStream zipInputStream = new ZipInputStream(new FileInputStream(serverZip));
+            ZipEntry nextEntry = null;
+            while ((nextEntry = zipInputStream.getNextEntry()) != null) {
+               String entryOutput = homePath.toString() + "/" + nextEntry.getName();
```
Comment (rvansa) [inline plugins/infinispan52/src/main/java/org/radargun/service/InfinispanServerService.java]: `File.separator`, or rather `homePath.resolve(nextEntry.getName)`.

File: plugins/infinispan52/src/main/java/org/radargun/service/InfinispanServerService.java [hunk 2]
```diff
@@ -89,6 +100,42 @@ public void init() {
...
+               if (nextEntry.isDirectory()) {
+                  Files.createDirectory(FileSystems.getDefault().getPath(entryOutput));
+               } else {
+                  FileOutputStream fout = new FileOutputStream(entryOutput);
+                  byte[] buffer = new byte[4096];
```
Comment (rvansa) [inline plugins/infinispan52/src/main/java/org/radargun/service/InfinispanServerService.java]: `Files.copy`

File: plugins/infinispan52/src/main/java/org/radargun/service/InfinispanServerService.java [hunk 3]
```diff
@@ -89,6 +100,42 @@ public void init() {
...
+                  zipInputStream.closeEntry();
+                  fout.close();
+               }
+            }
+
+            List<Path> homeContents = Files.list(homePath).collect(Collectors.toList());
+            // move the server files 1 directory up (remove the extra directory)
+            if (homeContents.size() == 1) {
+               Path fromDir = homeContents.get(0);
+               for (Path path : Files.list(fromDir).collect(Collectors.toList())) {
```
Comment (rvansa) [inline plugins/infinispan52/src/main/java/org/radargun/service/InfinispanServerService.java]: (minor stylistic) - `forEach`?

File: plugins/infinispan52/src/main/java/org/radargun/service/InfinispanServerService.java [hunk 4]
```diff
@@ -89,6 +100,42 @@ public void init() {
...
+                  Files.move(path, FileSystems.getDefault().getPath(home, path.getFileName().toString()));
+               }
+               Files.delete(fromDir);
+            }
+            // the extraction erases the executable bits
+            Path scriptPath = FileSystems.getDefault().getPath(home, "bin", getStartScriptPrefix() + (windows ? "bat" : "sh"));
+            Set<PosixFilePermission> permissions = PosixFilePermissions.fromString("rwxr-xr-x");
```
Comment (rvansa) [inline plugins/infinispan52/src/main/java/org/radargun/service/InfinispanServerService.java]: You should probably check  `getFileAttributeView(path, PosixFileAttributeView.class) != null` before this.

**Comment by jmarkos** [comment]
Thanks for the review, addressed all comments.

**Pull request merged**